src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IepLevelService.kt

```diff
@@ -40,9 +29,11 @@ class IepLevelService(
    * person’s level when they are transferred, we need to find the "nearest highest" level
    * according to HMPPS policy. Falls back to prison's default level when there's nothing else to do.
    */
-  suspend fun findNearestHighestLevel(prisonId: String, prisonLevels: List<IepLevel>, levelCode: String): String {
+  suspend fun findNearestHighestLevel(prisonId: String, prisonLevels: List<PrisonIncentiveLevel>, levelCode: String): String {
     val defaultLevelCode = chooseDefaultLevel(prisonId, prisonLevels)
-    val levelCodesAvailableInPrison = prisonLevels.filter(IepLevel::active).map(IepLevel::iepLevel).toSet()
+    val levelCodesAvailableInPrison = prisonLevels.filter(PrisonIncentiveLevel::active)
+      .map(PrisonIncentiveLevel::levelCode)
+      .toSet()
```

Comment:
Similar comment to above, if this service used the `PrisonIncentiveLevelService` it could easily get the available active levels and the default one (unless add that service to this would result in a cycle I'm not aware of, I didn't try)

Reply:
inactive levels aren't even passed into this method anymore. this entire class could be simplified a lot but i was wondering if that ought to be left for a slightly later pr once we're convinced that the data is all good in prod…?

now:
1) there will always be 1 default
2) there will not be inactive levels passed into either method in this class

previously, neither were guaranteed

Reply:
should hopefully be addressed by https://github.com/ministryofjustice/hmpps-incentives-api/pull/452